lib/publiccloud/utils.pm

```diff
@@ -233,12 +233,10 @@ sub register_addons_in_pc {
     my ($instance) = @_;
     my @addons = split(/,/, get_var('SCC_ADDONS', ''));
     my $remote = $instance->username . '@' . $instance->public_ip;
-    # Workaround for bsc#1245220
-    my $env = is_sle("=15-SP3") ? "ZYPP_CURL2=1" : "";
-    my $cmd = "sudo $env zypper -n --gpg-auto-import-keys ref";
-    my $ret = $instance->ssh_script_run(cmd => $cmd, timeout => 300);
+    my $zcmd = "--gpg-auto-import-keys ref";
+    my $ret = $instance->zypper_call_remote(cmd => $zcmd, exitcode => [0, 6], timeout => 300);
     die 'No enabled repos defined: bsc#1245651' if $ret == 6;    # from zypper man page: ZYPPER_EXIT_NO_REPOS
-    $instance->ssh_script_retry(cmd => $cmd, timeout => 300, retry => 3, delay => 120);
+    $instance->zypper_call_remote(cmd => $zcmd, timeout => 300, retry => 3, delay => 120);
```

Comment:
```suggestion
    $instance->zypper_call_remote(cmd => $zcmd, timeout => 300, retry => 3, delay => 1200);
```

Reply:
ok maybe 20 minutes was too much , but we often have empty repos on refresh just after registration and I have gut feeling that this is just because logic which suppose propagate repos after registration is too slow , so let's give it some time ... 5 minutes ?